Use pre-commit in CI (#4817)
ComputationalRadiationPhysics/picongpu: docs/COMMIT.md

From now on, each commit in this clone of the repo will be checked. See [pre-commit](https://pre-commit.com/) for usage
details. Some hints:

- You can run all hooks on all files via `pre-commit run --all-files [--hook-stage manual]`. The last argument
  `--hook-stage manual` includes rather slow additional tests that are run by the CI but are considered too heavy-weight
  to run before each commit.

> - If a check fails, oftentimes it will automatically apply the fixes (e.g. run `clang-format`). If there are no unstaged

Reviewer: optional (not part of this PR): 
```suggestion
- If a check fails, the fixes are often applied automatically (e.g., run `clang-format`). If there are no unstaged
```
